script.js

```diff
@@ -107,25 +110,37 @@ const applyToCells = callback => {
   }
 };
 
+//Function to implement the Color Selector
+let colorSelector = document.getElementById("color_choose"), color_change;
+const colorSelect = () => {
+  color_change = colorSelector.value;
+};
+
 // FILL UNCOLORED ------------------------
 const uncoloredFill = () => {
   applyToCells(col => {
     // check to see if color was set for cell
-    if (col.style.backgroundColor == "") {
+    if (col.style.backgroundColor == "" || col.style.backgroundColor == "white") {
       // if no color was set, set to currently selected color
-      col.style.backgroundColor = color;
+      col.style.backgroundColor = color_change;
     }
   });
 };
 
 // FILL ALL ------------------------------
 const fill = () => {
   // set to currently selected color regardless of prev color
-  applyToCells(col => (col.style.backgroundColor = color));
+  applyToCells(col => (col.style.backgroundColor = color_change));
 };
 
 // CLEAR ALL -----------------------------
 const clearAll = () => {
   // clear color regardless of prev color
   applyToCells(col => (col.style.backgroundColor = ""));
 };
+
+//Function to color the cell on user prompt
+function colorChange(){
```

Comment:
Function hoisting works, however it's a bit confusing. It should be preferred to move the logic and logic that it depends on to the top of the file, before the logic that depends on it. You can keep this if you'd like though.